instance/instance.go

```diff
@@ -32,9 +33,15 @@ const (
 
 //GetList a list of all instances
 func (s *Store) GetList(w http.ResponseWriter, r *http.Request) {
-	stateFilter := r.URL.Query().Get("state")
+	stateFilterQuery := r.URL.Query().Get("state")
+	var stateFilterList []string
+	if stateFilterQuery == "" {
```

Comment:
if query != "" { do stuff } - removes the need for else :) you don't need to set the list to nil, it's already nil from the way you created it

Reply:
Update this if/else statement to be the following:

```
if stateFilterQuery != "" {
  stateFilterList = strings.Split(stateFilterQuery, ",")
}
```